staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
That's a clever move. I would not have thought of it, because I don't think of this procedure as belonging to either the function or the `sync.WaitGroup`, but it reads nicely.

The only downside is that if a caller has a `sync.WaitGroup` in hand—and not a `wait.Group`—then he can't use these methods, right?